group-coordinator/src/main/java/org/apache/kafka/coordinator/group/streams/StreamsGroup.java

```diff
@@ -355,6 +355,32 @@ public StreamsGroupMember getOrMaybeCreateMember(
         return new StreamsGroupMember.Builder(memberId).build();
     }
 
+    /**
+     * Gets or creates a new member but without adding it to the group. Adding a member is done via the
+     * {@link StreamsGroup#updateMember(StreamsGroupMember)} method.
+     *
+     * @param memberId          The member ID.
+     * @param createIfNotExists Booleans indicating whether the member must be created if it does not exist.
+     * @return A StreamsGroupMember.
+     */
+    public StreamsGroupMember getOrMaybeCreateDefaultMember(
+        String memberId,
+        boolean createIfNotExists
+    ) throws UnknownMemberIdException {
+        StreamsGroupMember member = members.get(memberId);
+        if (member != null) {
+            return member;
+        }
+
+        if (!createIfNotExists) {
+            throw new UnknownMemberIdException(
+                String.format("Member %s is not a member of group %s.", memberId, groupId)
+            );
+        }
+
+        return StreamsGroupMember.Builder.withDefaults(memberId).build();
```

Comment:
The heartbeat relies on having various defaults (such as empty sets for tasksWithRevocations), that got replaced by null in a refactoring. That's why I had to introduce `withDefaults`.